### PR TITLE
Set containingClass at MethodParameter

### PR DESCRIPTION
**Pull request by mshima**
Fix spring doc failing to determine generic type due to lack of containing class information.

Related to https://github.com/jhipster/generator-jhipster/issues/17776.

### CODE REVIEW

File: springdoc-openapi-webmvc-core/src/test/resources/results/app181.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "*/*": {
+                "schema": {
+                  "type": "string"
```
Comment (mshima) [inline springdoc-openapi-webmvc-core/src/test/resources/results/app181.json:54]: `"type": "object"` is returned without this PR.

File: springdoc-openapi-webmvc-core/src/test/resources/results/app181.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+              "type": "integer",
+              "format": "int32"
+            }
+          },
+          {
+            "name": "genericField",
+            "in": "query",
+            "required": false,
+            "schema": {
+              "type": "string"
```
Comment (mshima) [inline springdoc-openapi-webmvc-core/src/test/resources/results/app181.json:44]: `"type": "object"` is returned without this PR.

**Comment by bnasslahsen** [comment]
@mshima, 

Thank you for your contribution to the project!

**Comment by bnasslahsen** [comment]
@mshima, 

I have made a little change 3061086aca2b8f4884473eb07cb8db1711db8a47 to keep the compatibility with earlier spring versions! (We still have demands to support spring-boot 1 for example)

**Comment by mshima** [comment]
@bnasslahsen no problem and thanks for the quick merge.
By the way, is there a plan for next release?
We are planning a new JHipster release this week.
Would be nice to have this fix, not blocker.

**Comment by bnasslahsen** [comment]
@mshima, 

I can confirm `v1.6.6` is now released!

**Comment by mshima** [comment]
@bnasslahsen thanks.

**Pull request merged**